spirv_glsl.cpp

```diff
@@ -3414,42 +3457,65 @@ std::pair<std::string, uint32_t> CompilerGLSL::flattened_access_chain_offset(uin
 		// Arrays
 		if (!type->array.empty())
 		{
-			// We have to modify the type, so keep a local copy.
-			if (&temp != type)
-				temp = *type;
-			type = &temp;
-
-			uint32_t array_size = temp.array.back();
-			temp.array.pop_back();
+			uint32_t array_stride = get_decoration(current_type, DecorationArrayStride);
+			if (!array_stride)
+				SPIRV_CROSS_THROW("SPIR-V does not define ArrayStride for buffer block.");
 
-			assert(type_size > 0);
-			assert(type_size % array_size == 0);
-
-			uint32_t array_stride = type_size / array_size;
-			assert(array_stride % 16 == 0);
+			auto *constant = maybe_get<SPIRConstant>(index);
+			if (constant)
+			{
+				// Constant array access.
+				offset += constant->scalar() * array_stride;
+			}
+			else
+			{
+				// Dynamic array access.
+				// FIXME: This will need to change if we support other flattening types than 32-bit.
+				const uint32_t word_stride = 16;
+				if (array_stride % word_stride)
+				{
+					SPIRV_CROSS_THROW(
+					    "Array stride for dynamic indexing must be divisible by the size of a 4-component vector. "
+					    "Likely culprit here is a float or vec2 array inside a push constant block. This cannot be "
```

Comment:
Are push constant blocks std430 or something along these lines? If it's possible to specify std140 layout for them we can note that in the error.